specs/spring-cloud-contract-spec-kotlin/src/main/kotlin/org/springframework/cloud/contract/spec/internal/FromRequestDsl.kt

```diff
@@ -0,0 +1,173 @@
+
+	/**
+	 * @return URL path and query.
+	 */
+	fun url() = FromRequestDslProperty(template.escapedUrl())
+
+	/**
+	 * First value of a query parameter e.g. request.query.search.
+	 * @param key key for the query param
+	 * @return dsl property
+	 */
+	fun query(key: String) = FromRequestDslProperty(template.escapedQuery(key))
+
+	/**
+	 * nth value of a query parameter (zero indexed) e.g. request.query.search.[5].
+	 * @param key key for the query param
+	 * @param index index of the query param
+	 * @return dsl property
+	 */
+	fun query(key: String, index: Int) = FromRequestDslProperty(template.escapedQuery(key, index))
+
+	/**
+	 * URL path.
+	 * @return dsl property
+	 */
+	fun path() = FromRequestDslProperty(template.escapedPath())
+
+	/**
+	 * nth value of a URL path (zero indexed) e.g. {{{ request.path.[2] }}}* @param index.
+	 * @param index path index
+	 * @return dsl property
+	 */
+	fun path(index: Int) = FromRequestDslProperty(template.escapedPath(index))
+
+	/**
+	 * First value of a request header e.g. request.headers.X-Request-Id.
+	 * @param key header key
+	 * @return dsl property
+	 */
+	fun header(key: String) = FromRequestDslProperty(template.escapedHeader(key))
+
+	/**
+	 * nth value of a request header (zero indexed) e.g. request.headers.X-Request-Id.
+	 * @param key header key
+	 * @param index header index
+	 * @return dsl property
+	 */
+	fun header(key: String, index: Int) = FromRequestDslProperty(template.escapedHeader(key, index))
+
+	/**
+	 * Retruns the tempalte for retrieving the first value of a cookie with certain key.
+	 * @param key cookie key
+	 * @return dsl property
+	 */
+	fun cookie(key: String) = FromRequestDslProperty(template.escapedCookie(key))
+
+	/**
+	 * Request body text (avoid for non-text bodies).
+	 * @return dsl property
+	 */
+	fun body() = FromRequestDslProperty(template.escapedBody())
+
+	/**
+	 * Request body text for the given JsonPath.
+	 * @param jsonPath json path body
+	 * @return dsl property
+	 */
+	fun body(jsonPath: String) = FromRequestDslProperty(template.escapedBody(jsonPath))
+
+	/**
+	 * Unescaped URL path and query.
+	 * @return dsl property
+	 */
+	fun rawUrl() = FromRequestDslProperty(template.url())
+
+	/**
+	 * Unescaped First value of a query parameter e.g. request.query.search.
+	 * @param key query key
+	 * @return dsl property
+	 */
+	fun rawQuery(key: String) = FromRequestDslProperty(template.query(key))
+
+	/**
+	 * Unescaped nth value of a query parameter (zero indexed) e.g.
+	 * request.query.search.[5].
+	 * @param key query key
+	 * @param index query index
+	 * @return dsl property
+	 */
+	fun rawQuery(key: String, index: Int) = FromRequestDslProperty(template.query(key, index))
+
+	/**
+	 * Unescaped URL path.
+	 * @return dsl property
+	 */
+	fun rawPath() = FromRequestDslProperty(template.path())
+
+	/**
+	 * Unescaped nth value of a URL path (zero indexed) e.g. {{{ request.path.[2]. }}}*
```

Comment:
Add full stop at the end.

Reply:
Ah, the checkstyle rules from Boot are not applicable for Kotlin files. That's a shame.